app/controllers/issue_templates_controller.rb

```diff
@@ -1,206 +1,200 @@
+                except: [:preview, :move_order_higher, :move_order_lower, :move_order_to_top, :move_order_to_bottom, :move]
+  before_filter :find_tracker, only: [:set_pulldown]
+
+  def index
+    tracker_ids = IssueTemplate.where('project_id = ?', @project.id).pluck(:tracker_id)
+
+    @template_map = {}
+    tracker_ids.each do |tracker_id|
+      templates = IssueTemplate.where('project_id = ? AND tracker_id = ?',
+                                      @project.id, tracker_id).order('position')
+      @template_map[Tracker.find(tracker_id)] = templates if templates.any?
+    end
+
+    @issue_templates = IssueTemplate.where('project_id = ?',
+                                           @project.id).order('position')
+
+    @setting = IssueTemplateSetting.find_or_create(@project.id)
+    inherit_template = @setting.enabled_inherit_templates?
+    @inherit_templates = []
+
+    project_ids = inherit_template ? @project.ancestors.collect(&:id) : [@project.id]
+    if inherit_template
+      # keep ordering
+      used_tracker_ids = @project.trackers.pluck(:tracker_id)
+
+      project_ids.each do |i|
+        @inherit_templates.concat(IssueTemplate.where('project_id = ? AND enabled = ?
+          AND enabled_sharing = ? AND tracker_id IN (?)', i, true, true, used_tracker_ids).order('position'))
+      end
+    end
+
+    @global_issue_templates = GlobalIssueTemplate.joins(:projects).where(['projects.id = ?', @project.id]).order('position')
+
+    render layout: !request.xhr?
+  end
+
+  def show
+  end
+
+  def new
+    # create empty instance
+    @issue_template ||= IssueTemplate.new(author: @user, project: @project)
+    if request.post?
+      @issue_template.safe_attributes = params[:issue_template]
+      if @issue_template.save
+        flash[:notice] = l(:notice_successful_create)
+        redirect_to action: 'show', id: @issue_template.id, project_id: @project
+      end
+    end
+  end
+
+  def edit
+    # Change from request.post to request.patch for Rails4.
+    if request.patch? || request.put?
+      @issue_template.safe_attributes = params[:issue_template]
+      if @issue_template.save
+        flash[:notice] = l(:notice_successful_update)
+        redirect_to action: 'show', id: @issue_template.id, project_id: @project
+      end
+    end
+  end
+
+  def destroy
+    if request.post?
+      if @issue_template.destroy
+        flash[:notice] = l(:notice_successful_delete)
+        redirect_to action: 'index', project_id: @project
+      end
+    end
+  end
+
+  # load template description
+  def load
+    @issue_template = if !params[:template_type].nil? && params[:template_type] == 'global'
+                        GlobalIssueTemplate.find(params[:issue_template])
+                      else
+                        IssueTemplate.find(params[:issue_template])
+                      end
+    render text: @issue_template.to_json(root: true)
+  end
+
+  # update pulldown
+  def set_pulldown
```

Comment:
__[rails_best_practices]__
* move model logic into model (group use_count > 4) [:ref](http://rails-bestpractices.com/posts/2010/07/21/move-model-logic-into-the-model/)
* move model logic into model (@project use_count > 4) [:ref](http://rails-bestpractices.com/posts/2010/07/21/move-model-logic-into-the-model/)
* move model logic into model (@tracker use_count > 4) [:ref](http://rails-bestpractices.com/posts/2010/07/21/move-model-logic-into-the-model/)
* move model logic into model (x use_count > 4) [:ref](http://rails-bestpractices.com/posts/2010/07/21/move-model-logic-into-the-model/)